index.rst

```diff
@@ -1,25 +1,429 @@
+It allows a new node to obtain the current state of the system without issuing commands to the running controllers.
+
+Differences Between DDS and Kafka
+=================================
+
+Kafka is a broker-based system, whereas DDS is distributed, with no central broker.
+
+With DDS each node (e.g. CSC or Jupyter Notebook) discovers the other nodes, using UDP, as the software starts up.
+Reliability is achieved by switching which node acts as “master” if one fails.
+
+With Kafka each node connects to a specified broker and schema registry, in order to read and write messages.
+There is no discovery phase and no need for UDP.
+Reliability and performance can be increased by running multiple brokers in parallel, and by using more than one "partition" for each topic.
+
+Schema Evolution
+----------------
+
+DDS does not support schema evolution, at least as we are using it.
+When we update the schema for a topic it becomes an entirely new topic.
+Internally the topic has a different name; the topic names DDS sees include a hash that represents the schema
+though we hide this hash from users.
+
+Kafka schemas support basic schema evolution: we may add or delete fields and readers and writers can freely use older or newer schemas.
+Note that even with this type of schema evolution, we will have to reset part of the system if we change the type of a message field.
+
+Licensing
+---------
+
+Kafka is open-source, ADLink OpenSplice is not.
+With Kafka developers can run the full system, with all features, without licensing headaches.
+With ADLink OpenSplice we presently run two very different versions of the software, in order to reduce costs and avoid licensing headaches:
+
+* Developers use the “community” edition, which ADLink no longer offers, and which only supports single-process topic daemons.
+
+* Deployment uses the commercial edition, because it supports shared-memory topic daemons, which work better for large systems.
+  The commerical version is also required to run ADLink's diagnostic tools.
+  Note that the commercial edition has always been newer than the community edition.
+  In the past ADLink kept the community edition one major version behind the commercial edition.
+  But the last community version they served is now two major versions behind the commercial edition.
+
+Commercial support for Kafka is available from Confluent, should we need it.
+
+The Engineering Facilities Database
+-----------------------------------
+
+DDS requires running complex extra software (ts_salkafka, an in-house package) in order to copy all messages from DDS to Kafka, so we can ingest them into the Engineering Facilities Database (EFD).
+If we use Kafka then we can eliminate this extra software.
+
+Performance
+===========
+
+I measured performance by writing 10,000 messages for a specified topic in one process, and reading the same in another.
```

Comment:
I'm not keen on 3rd person (having been brought up with Strunk and White's enthusiasm for active voice), but agree that "I" seems out of place. I made this bit 3rd person and changed the rest to either use 3rd person or "we" instead of "I".